SimpleWeather/DailyWeatherForecast/DailyWeatherForecastViewController.swift

```diff
@@ -0,0 +1,236 @@
+    @IBOutlet weak var currentWeatherIconImageView: UIImageView!
+    @IBOutlet weak var currentWeatherTemperatureLabel: UILabel!
+    @IBOutlet weak var currentWeatherCityNameLabel: UILabel!
+    @IBOutlet weak var dailyHourlyForecastCollectionView: UICollectionView!
+    @IBOutlet weak var windLabel: UILabel!
+    @IBOutlet weak var pressureLabel: UILabel!
+    @IBOutlet weak var humidityLabel: UILabel!
+    @IBOutlet weak var windDegLabel: UILabel!
+    
+    // MARK: Properties
+    
+    var weatherForecast: WeatherForecast?
+    var presenter: DailyWeatherForecastPresenterProtocol!
+    
+    var isContentLoaded = false
+    var isDashedCircleViewAnimationNeedShow = true
+    
+    var sunrise = Date()
+    var sunset = Date()
+    var currentTime = Date()
+    var timeZone = 0
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        setupCollectionView()
+        setupRefreshControl()
+        presenter.viewDidLoad()
+        showLoading()
+    }
+    
+    // MARK: UI setup
+    
+    func showLoading() {
+        view.subviews.forEach { $0.isHidden = !$0.isHidden }
+        activityIndicatorView.isHidden = false
+        activityIndicatorView.startAnimating()
+        tabBarController?.tabBar.isHidden = true
+    }
+    
+    func showContent() {
+        isContentLoaded = true
+        view.subviews.forEach { $0.isHidden = !$0.isHidden }
+        activityIndicatorView.isHidden = true
+        activityIndicatorView.stopAnimating()
+        tabBarController?.tabBar.isHidden = false
+    }
+    
+    func showLocationAccessHint(_ show: Bool) {
+        locationAccessHintLabel.isHidden = !show
+    }
+    
+    func setupRefreshControl() {
+        scrollView.refreshControl = UIRefreshControl()
+        scrollView.refreshControl?.addTarget(self, action: #selector(updateWeather), for: .valueChanged)
+    }
+    
+    func setupCollectionView() {
+        let flowLayout = dailyHourlyForecastCollectionView.collectionViewLayout as! UICollectionViewFlowLayout
+        flowLayout.sectionInset = UIEdgeInsets(top: 0, left: 24, bottom: 0, right: 24)
+        flowLayout.minimumLineSpacing = 48
+    }
+    
+    @objc func updateWeather() {
+        presenter.updateWeather()
+    }
+    
+    func isDashedCircleViewVisible(view: UIView) -> Bool {
+        
+        func isVisible(view: UIView, inView: UIView?) -> Bool {
+            
+            guard let inView = inView else { return true }
+            
+            let viewFrame = inView.convert(view.bounds, from: view)
+            if viewFrame.intersects(inView.bounds) {
+                return isVisible(view: view, inView: inView.superview)
+            }
+            
+            return false
+        }
+        
+        return isVisible(view: view, inView: view.superview)
+    }
+    
+    func getCurrentTime() -> Date {
+        return NSDate() as Date
+    }
+    
+    func checkIfTimeInRange(_ currentTime: Date) -> Bool {
```

Comment:
я бы сделал extension на Date
```suggestion
    func checkIfDateInRange(start: Date, end: Date) -> Bool {
```